### PR TITLE
Loom (1.18)

### PR DESCRIPTION
**Pull request by Charnuz**

### CODE REVIEW

File: src/main/java/net/dries007/tfc/client/render/LoomBlockEntityRenderer.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+import net.dries007.tfc.common.blocks.wood.TFCLoomBlock;
+
+public class LoomBlockEntityRenderer implements BlockEntityRenderer<LoomBlockEntity>
+{
+    public LoomBlockEntityRenderer(BlockEntityRendererProvider.Context context)
+    {
+    }
+
+    @Override
+    public void render(LoomBlockEntity te, float partialTicks, PoseStack matrixStack, MultiBufferSource buffer, int combinedLight, int combinedOverlay)
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/client/render/LoomBlockEntityRenderer.java]: param name should be loom, not te

File: src/main/java/net/dries007/tfc/client/render/LoomBlockEntityRenderer.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+import com.mojang.math.Matrix3f;
+import com.mojang.math.Matrix4f;
+import com.mojang.math.Vector3f;
+import net.dries007.tfc.client.ClientHelpers;
+import net.dries007.tfc.common.blockentities.LoomBlockEntity;
+import net.dries007.tfc.common.blocks.wood.TFCLoomBlock;
+
+public class LoomBlockEntityRenderer implements BlockEntityRenderer<LoomBlockEntity>
+{
+    public LoomBlockEntityRenderer(BlockEntityRendererProvider.Context context)
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/client/render/LoomBlockEntityRenderer.java]: I think you can register the renderer like `ctx -> new LoomBER()` and skip having a constructor entirely

File: src/main/java/net/dries007/tfc/client/render/LoomBlockEntityRenderer.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    {
+    }
+
+    @Override
+    public void render(LoomBlockEntity te, float partialTicks, PoseStack matrixStack, MultiBufferSource buffer, int combinedLight, int combinedOverlay)
+    {
+        if (!(te.getBlockState().getBlock() instanceof TFCLoomBlock)) return;
+
+        matrixStack.pushPose();
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/client/render/LoomBlockEntityRenderer.java]: poseStack

File: src/main/java/net/dries007/tfc/common/blocks/wood/TFCLoomBlock.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        {
+            case NORTH:
+            default:
+                return SHAPE_NORTH;
+            case SOUTH:
+                return SHAPE_SOUTH;
+            case WEST:
+                return SHAPE_WEST;
+            case EAST:
+                return SHAPE_EAST;
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/common/blocks/wood/TFCLoomBlock.java]: Let IDEA change this to an enhanced switch

File: src/main/java/net/dries007/tfc/common/blocks/wood/TFCLoomBlock.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        }
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public InteractionResult use(BlockState state, Level worldIn, BlockPos pos, Player player, InteractionHand handIn, BlockHitResult hit)
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/common/blocks/wood/TFCLoomBlock.java]: worldIn -> level
handIn -> hand

File: src/main/java/net/dries007/tfc/common/blocks/wood/TFCLoomBlock.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+    @Nullable
+    @Override
+    public BlockState getStateForPlacement(BlockPlaceContext context)
+    {
+        return defaultBlockState().setValue(FACING, context.getHorizontalDirection().getOpposite());
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> p_54794_)
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/common/blocks/wood/TFCLoomBlock.java]: unnamed param

File: src/main/java/net/dries007/tfc/common/blocks/wood/Wood.java [hunk 7]
```diff
@@ -25,16 +25,18 @@
...
 import net.dries007.tfc.common.blockentities.TFCBlockEntities;
 import net.dries007.tfc.common.blocks.ExtendedProperties;
 import net.dries007.tfc.common.blocks.GroundcoverBlock;
 import net.dries007.tfc.common.blocks.TFCBlocks;
 import net.dries007.tfc.common.blocks.devices.SluiceBlock;
 import net.dries007.tfc.common.items.ChestBlockItem;
-import net.dries007.tfc.util.Helpers;
 import net.dries007.tfc.world.feature.tree.TFCTreeGrower;
 
+import static net.dries007.tfc.util.Helpers.identifier;
```
Comment (eerussianguy) [inline src/main/java/net/dries007/tfc/common/blocks/wood/Wood.java]: Not really our code style to do this

**Review by eerussianguy — CHANGES_REQUESTED** [review]

File: src/main/java/net/dries007/tfc/client/render/LoomBlockEntityRenderer.java (hunk 2 above)
Comment (Charnuz) [reply src/main/java/net/dries007/tfc/client/render/LoomBlockEntityRenderer.java]: Turns out that was just a remainder from porting it from latejuly and isn't needed currently, though I can change that line still if you want to match the current style

**Pull request merged**